File: backend/app/schemas/department.py
from datetime import datetime

from pydantic import BaseModel, Field


class DepartmentCreateRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=120)
    code: str = Field(..., min_length=1, max_length=32)
    parent_id: int | None = None
    sort_order: int = 0
    status: str = "enabled"


class DepartmentUpdateRequest(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=120)
    code: str | None = Field(default=None, min_length=1, max_length=32)
    parent_id: int | None = None
    sort_order: int | None = None
    status: str | None = None


class DepartmentItem(BaseModel):
    id: int
    name: str
    code: str
    parent_id: int | None = None
    sort_order: int
    status: str
    created_at: datetime
    updated_at: datetime


class DepartmentTreeItem(DepartmentItem):
    children: list["DepartmentTreeItem"] = Field(default_factory=list)

File: frontend/src/api/department.ts
import { apiDelete, apiGet, apiPost, apiPut } from './request'

export interface DepartmentItem {
  id: number
  name: string
  code: string
  parent_id?: number | null
  sort_order: number
  status: string
  created_at: string
  updated_at: string
  children?: DepartmentItem[]
}

export function listDepartments() {
  return apiGet<DepartmentItem[]>('/departments')
}

export function listDepartmentTree() {
  return apiGet<DepartmentItem[]>('/departments/tree')
}

export function createDepartment(data: { name: string; code: string; parent_id?: number | null; sort_order?: number; status?: string }) {
  return apiPost<DepartmentItem>('/departments', data)
}

export function updateDepartment(id: number, data: { name?: string; code?: string; parent_id?: number | null; sort_order?: number; status?: string }) {
  return apiPut<DepartmentItem>(`/departments/${id}`, data)
}

export function deleteDepartment(id: number) {
  return apiDelete<{ deleted: boolean }>(`/departments/${id}`)
}
